app.py

```
@@ -506,41 +1014,45 @@
+                      low_attendance_query_params.append(course_id_int)
+                      selected_course_name = course_name_res['name'] # Assign the actual name
+                      print(f"Applying course filter: ID={course_id_int}, Name={selected_course_name}")
                  else:
-                      # Handle case where invalid course_id was passed
+                      # Handle case where invalid course_id was passed in URL params
                       selected_course_id = 'all' # Reset to 'all'
-                      selected_course_name = None # Keep it None if not found
-                      flash("Invalid course selected.", "warning") # Inform the user
+                      selected_course_name = "All Courses" # Reset display name
+                      flash("Invalid course ID provided in parameters for filtering.", "warning")
+                      print(f"Invalid course ID {course_id} provided for filtering low attendance.")
```

Ruff (0.8.2)

Undefined name `course_id`

(F821)